crates/lib/src/relation.rs

```diff
@@ -182,44 +187,41 @@ pub struct Header {
 
 impl From<Header> for ProductType {
     fn from(value: Header) -> Self {
-        ProductType::from_iter(value.fields.iter().map(|x| match &x.field {
-            FieldName::Name { field, .. } => ProductTypeElement::new_named(x.algebraic_type.clone(), field),
-            FieldName::Pos { .. } => ProductTypeElement::new(x.algebraic_type.clone(), None),
-        }))
+        ProductType::from_iter(
+            value
+                .fields
+                .into_iter()
+                .map(|x| ProductTypeElement::new(x.algebraic_type, x.field.into_field_name())),
+        )
     }
 }
 
 impl Header {
-    pub fn new(table_name: &str, fields: &[Column]) -> Self {
-        Self {
-            table_name: table_name.into(),
-            fields: fields.into(),
-        }
-    }
-
-    pub fn from_product_type(table_name: &str, fields: ProductType) -> Self {
-        let mut cols = Vec::with_capacity(fields.elements.len());
-
-        for (pos, f) in fields.elements.into_iter().enumerate() {
-            let name = match f.name {
-                None => FieldName::Pos {
-                    table: table_name.into(),
-                    field: pos,
-                },
-                Some(x) => FieldName::Name {
-                    table: table_name.into(),
-                    field: x,
-                },
-            };
-            cols.push(Column::new(name, f.algebraic_type));
-        }
+    pub fn new(table_name: String, fields: Vec<Column>) -> Self {
+        Self { table_name, fields }
+    }
+
+    pub fn from_product_type(table_name: String, fields: ProductType) -> Self {
+        let cols = fields
```

Comment:
I wonder if these iter chains hurt debug build performance more than the loop form on the left. Do you happen to have any knowledge of how that works @Centril?